OpenRA.Mods.Common/EditorBrushes/EditorDefaultBrush.cs

```diff
@@ -51,6 +52,15 @@ public EditorDefaultBrush(EditorViewportControllerWidget editorWidget, WorldRend
 				.ToDictionary(r => r.Info.ResourceType, r => r);
 		}
 
+		long CalculateActorSelectionPriority(EditorActorPreview actor)
+		{
+			var center = new int2(actor.Bounds.X, actor.Bounds.Y);
+			long priority = (center - worldPixel).Length;
+			priority <<= 32;
```

Comment:
Well all it needs to be is enough pixel closeness to dominate world height, `int`s are 32 bits, `long`s are 64 bits, and shifting 32 bits is the standard approach to stuffing two 32-bit words in a 64-bit word.